### ISSUE TITLE
Subsetting ADCP file by time producing error

### ISSUE THREAD

**Issue by cjpratt94** [dankelley/oce#1579]
For example:
```R
d<-read.oce("PER2018043_Sentinel-V_132_54_Ship-Harbour-Deep_M2067_May18-Nov18.pd0")

> ds <- subset(d, time<(as.POSIXct("2018-05-03 15:00:00", tz="UTC")), debug=3)
    subsetting an adp by time
    keeping bins:: TRUE, TRUE, ..., FALSE, FALSE (length 134865)
    number of kept bins: 720 
    subsetting x@data$v, which is an array
Error in x@data[[name]][keep, , , drop = FALSE] : 
  (subscript) logical subscript too long
```

**Comment by dankelley** [comment]
FYI (to other developers): I have a private data file to test this.

**Comment by dankelley** [comment]
I can confirm the problem, as shown below.  Notes:
1. it is setting `by=6` because the file is large. I'll do another test with `by=1`
2. the `time` vector is 1 longer than it ought to be, compared with the other data (134865 vs 134864)

```R
> library(oce)
> d<-read.oce("/Volumes/GLIDER_DATA/PER2018043_Sentinel-V_132_54_Ship-Harbour-Deep_M2067_May18-Nov18.pd0")
setting by=6 for a large RDI file
setting monitor=TRUE for a large RDI file
Increasing ensembles,times,sec100s storage to 150000 elements ...
  |                                                                               |   0%
  |===============================================================================| 100%
Finished reading 134864 profiles from "/Volumes/GLIDER_DATA/PER2018043_Sentinel-V_132_54_Ship-Harbour-Deep_M2067_May18-Nov18.pd0"
Warning messages:
1: In read.adp.rdi(file, processingLog = processingLog, ...) :
  setting by=6 for a large RDI file

2: In read.adp.rdi(file, processingLog = processingLog, ...) :
  setting monitor=TRUE for a large RDI file

3: In read.adp.rdi(file, processingLog = processingLog, ...) :
  skipping the first ensemble (a temporary solution that eases reading of SentinelV files)

4: In read.adp.rdi(file, processingLog = processingLog, ...) :
  A list of unhandled segment codes follows. Several Teledyne RDI manuals
  describe such codes; see e.g. Table 33 of Teledyne RD Instruments, 2014.
  Ocean Surveyor/Ocean Observer Technical Manual.
  P/N 95A-6012-00 April 2014 (OS_TM_Apr14.pdf)
    Code 0x01 0x0f occurred 134865 times
    Code 0x00 0x70 occurred 134865 times
    Code 0x01 0x70 occurred 134865 times
    Code 0x02 0x70 occurred 134865 times
    Code 0x00 0x32 occurred 134865 times
    Code 0x04 0x70 occurred 134865 times

> str(d@data,1)
List of 31
 $ v                  : num [1:134864, 1:59, 1:4] 0.766 -0.092 -0.182 -0.297 -0.325 -0.872 -0.553 0.12 -0.213 -0.831 ...
 $ q                  : raw [1:134864, 1:59, 1:4] 45 37 33 3e ...
 $ a                  : raw [1:134864, 1:59, 1:4] 44 43 44 43 ...
 $ g                  : NULL
 $ vv                 : num [1:134864, 1:59] -0.56 -0.636 -1.111 -0.139 -0.438 ...
 $ vq                 : raw [1:134864, 1:59] 2f 2f 20 5e ...
 $ va                 : raw [1:134864, 1:59] 57 58 57 58 ...
 $ vg                 : NULL
 $ vdistance          : num [1:59] 3.33 4.33 5.33 6.33 7.33 ...
 $ distance           : num [1:59] 3.3 4.3 5.3 6.3 7.3 8.3 9.3 10.3 11.3 12.3 ...
 $ time               : POSIXct[1:134865], format: "2018-05-02 15:00:00" "2018-05-02 15:00:12" ...
 $ pressure           : num [1:134864] 0.097 0.098 0.097 0.096 0.097 0.098 0.098 0.101 0.098 0.1 ...
 $ temperature        : num [1:134864] 31.7 31.7 31.8 31.8 31.8 ...
 $ salinity           : int [1:134864] 32 32 32 32 32 32 32 32 32 32 ...
 $ depth              : num [1:134864] 0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1 ...
 $ soundSpeed         : int [1:134864] 1546 1546 1546 1546 1546 1546 1546 1546 1546 1546 ...
 $ heading            : num [1:134864] 116 116 116 116 116 ...
 $ pitch              : num [1:134864] -4.87 -4.88 -4.87 -4.86 -4.88 ...
 $ roll               : num [1:134864] -1.97 -1.96 -1.98 -1.94 -1.93 -1.91 -1.9 -1.9 -1.96 -1.93 ...
 $ headingStd         : num [1:134864] 0 0 0 0 0 0 0 0 0 0 ...
 $ pitchStd           : num [1:134864] 0 0 0 0 0 0 0 0 0 0 ...
 $ rollStd            : num [1:134864] 0 0 0 0 0 0 0 0 0 0 ...
 $ pressureStd        : int [1:134864] 0 0 0 0 0 0 0 0 0 0 ...
 $ xmitCurrent        : num [1:134864] 0 0 0 0 0 0 0 0 0 0 ...
 $ xmitVoltage        : num [1:134864] 191 191 191 191 191 191 191 191 191 191 ...
 $ ambientTemp        : num [1:134864] 0 0 0 0 0 0 0 0 0 0 ...
 $ pressurePlus       : num [1:134864] 0 0 0 0 0 0 0 0 0 0 ...
 $ pressureMinus      : num [1:134864] 0 0 0 0 0 0 0 0 0 0 ...
 $ attitudeTemp       : num [1:134864] 0 0 0 0 0 0 0 0 0 0 ...
 $ attitude           : num [1:134864] 0 0 0 0 0 0 0 0 0 0 ...
 $ contaminationSensor: num [1:134864] 0 0 0 0 0 0 0 0 0 0 ...
```

**Comment by dankelley** [comment]
Using
```R
d<-read.oce("/Volumes/GLIDER_DATA/PER2018043_Sentinel-V_132_54_Ship-Harbour-Deep_M2067_May18-Nov18.pd0",by=1)
```
gives as follows (a snippet of `str` output), so this matter of an extra time is not related to the `by` argument.
```R
.. ..$ time               : POSIXct[1:809189], format: "2018-05-02 15:00:00" "2018-05-02 15:00:02" "2018-05-02 15:00:04" ...
  .. ..$ pressure           : num [1:809188] 0.097 0.098 0.097 0.099 0.097 0.097 0.097 0.098 0.098 0.096 ...
 
```

**Comment by dankelley** [comment]
This is fixed now in the "develop" branch commit 21f82809b7c498297e8a0e540f3a4d4e2abed592

@cjpratt94 -- if your system is set to build R packages from source, you ought to be able to install the new version by typing the following into an R console. (Obviously, you'll need to `install.packages("devtools")` if that is not already installed.)
```R
library(devtools)
install_github("dankelley/oce", ref="develop")
```

There are some instructions for setting up to build packages at https://github.com/dankelley/oce/wiki/Configuring-a-system-to-be-able-to-build-oce-from-source

If you feel you have the time and ability to test this out, please do so and consider closing the issue. In the oce project, the convention is that reporters close issues. That prevents the developers from ignoring users' needs. However, if it's a hassle for you to set up to install the oce update, just say so, and the developers will coordinate on whether to close the issue.

Thanks again for the issue report!

**Comment by dankelley** [comment]
This issue has been closed, since it seems to have been addressed, and since  discussion has stalled. The original reporter should feel free to reopen it, though, or to open new issues that might be related. Thanks.

PS.  This is a standardized reply.
